homeassistant/components/intellifire/sensor.py

```diff
@@ -126,3 +94,40 @@ async def async_setup_entry(
         value_fn=_time_remaining_to_timestamp,
     ),
 )
+
+
+class IntellifireSensor(
+    IntellifireEntity, SensorEntity, IntellifireSensorRequiredKeysMixin
+):
+    """Extends IntellifireEntity with Sensor specific logic."""
+
+    def __init__(
+        self,
+        coordinator: IntellifireDataUpdateCoordinator,
+        description: IntellifireSensorEntityDescription,
+    ) -> None:
+        """Init Function with really bad docstring."""
+        super().__init__(coordinator=coordinator, description=description)
+
+        self.description = description
+
+    @property
+    def native_value(self) -> int | str | datetime | None:
+        """Return the state."""
+
+        # This used to be self.entity_description -> but had to use
+        # description in order to access value_fn function
+        # may be a better way to do this
+        return self.description.value_fn(self.coordinator.api.data)
```

Comment:
The old version was referring to `self.entity_description.value_fn` but I was getting an error that `entity_description` didn't have `value_fn` -> So earlier I assigned `self.description = description` ... I don't like this but got stuck on figuring out how to actually make it work the old way